openssl/src/ossl_param.rs

```diff
@@ -0,0 +1,182 @@
+//! OSSL_PARAM management for OpenSSL 3.*
+//!
+//! The OSSL_PARAM structure represents generic attribute that can represent various
+//! properties in OpenSSL, including keys and operations.
+//!
+//! For convinience, the OSSL_PARAM_BLD builder can be used to dynamically construct
+//! these structure.
+//!
+//! Note, that this module is available only in OpenSSL 3.* and
+//! only internally for this crate!
+
+// Depending on which version of OpenSSL is used, and which algorithms
+// are exposed in the bindings, not all of these functions are used.
+#![allow(dead_code)]
```

Comment:
Let's just mark the methods with the appropriate cfgs, I don't like missing out on dead code in general